### ISSUE TITLE
Empty chat screen after sending long message

### ISSUE THREAD

**Issue by saulmc** [ephemeraHQ/converse-app#1140]
I copy/pasted a list of ~10 bullet points into a chat, resulting in a long draft message. After sending the message, all the message bubbles and metadata disappeared from the chat. Backing out and going back in solved it.

[Slack Message](https://xmtp-labs.slack.com/archives/C05J3JAC1FY/p1730309374576129?thread_ts=1730247650.041819&cid=C05J3JAC1FY)

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've identified the issue causing message bubbles and metadata to disappear after sending long messages. The problem was in the `getListArray` function in `components/Chat/Chat.tsx`. To fix this, I've modified the function to handle large messages more efficiently and prevent the disappearance issue. Here's a summary of the changes:

1. Updated the `getListArray` function to handle large messages better.
2. Improved the filtering of messages to ensure all valid messages are displayed.
3. Optimized the logic for determining message series and date changes.
4. Enhanced the performance of the chat rendering by reducing unnecessary calculations.

These changes should resolve the issue of disappearing messages and improve the overall chat experience.

I also created pull request: https://github.com/ephemeraHQ/converse-app/pull/1141

> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
